CMakeLists.txt

```diff
@@ -17,16 +17,22 @@ include(CTest)
 
 find_program(YAML_CPP_CLANG_FORMAT_EXE NAMES clang-format)
 
+if (NOT DEFINED YAML_CPP_MASTER_PROJECT)
```

Comment:
Is this pattern common for CMake? It seems like you're trying to solve a pretty common problem.

Reply:
I'm very curious on why a number of very well known libraries such as this one, libfmt, spdlog, etc) follow the pattern of not automatically creating the install targets when they're not the top level project.  I guess I don't understand the use case or when that would be useful. Could you shed some light on that?

Reply:
The big reason the fix is needed is that the current implementation breaks the install ability in projects that include this via CMake's FetchContent. (CMake complains target xxx requires yaml-cpp which is not going to be installed (or something to that effect)